TakingITGlobal/commit2act
# [TIG-587] ESLint auto-fix and top-level overrides removed

## Ticket

* [TIG-587](https://app.clickup.com/t/9009201449/TIG-587)

## Description

This PR removes the error-to-warning overrides at the root `.eslintrc.json` config, updates the simpler (and automatic via `--fix`) issues, and implements the overrides only in the `pages` and `components` sub-directories - where the remaining issues occur. 

This also introduces a `prop-types/component.js` file with basic extendable `BaseComponent` and `LinkComponent` prop type objects to make mocking out each component's props simpler. 

## Testing

* Pull down and build the branch locally
* Run `npm run lint` 
* Confirm no errors show up in the issues summary
* Confirm all warnings are from the `pages` and `components` folders
* Run `npm start`
* Wait until app builds and serves at `http://localhost:3000/`
* Login and navigate around, keeping an eye on the dev console
* Confirm no issues occur or existing functionality has changed

## Screenshots

**Linter output down to 418 warnings (mostly errors) localized under pages and components:**

![image](https://github.com/TakingITGlobal/commit2act/assets/1672105/5c784d50-b93a-4d2d-93b6-d9d68f3888ae)
